## Aim

The aim of this notebook is to check the availability of venue info. 

In [54]:
import pandas as pd

In [63]:
REF_DF = pd.read_csv('../../data/processed/openalex_reference_paper_df_unique.csv')

In [111]:
CIT_DF = pd.read_csv('../../data/processed/large/openalex_citation_concept_df.csv')
CIT_ID_NAME = 'Citation Paper OpenAlex ID'

In [112]:
CIT_DF[CIT_DF[CIT_ID_NAME] == 'W3211836602']

Empty DataFrame
Columns: [Cited Ppaer Year, Cited Paper DOI, Cited Paper Title, Cited Paper OpenAlex ID, Citation Paper Year, Citation Paper OpenAlex ID, Citation Ppaer OpenAlex Title, Citation Paper OpenAlex DOI, Number of Concepts, Index of Concept, Concept, Concept ID, Wikidata, Level, Score]
Index: []

In [76]:
CIT_DF = CIT_DF[CIT_DF[CIT_ID_NAME].notnull()]

In [77]:
cit_unique = CIT_DF.iloc[:, 4:].drop_duplicates()

In [78]:
cit_ids = cit_unique['Citation Paper OpenAlex ID'].tolist()
len(list(set(cit_ids)))

60272

In [108]:
CIT_DF_ORIGINAL = pd.read_csv('../../data/processed/openalex_citation_paper_df.csv')

In [88]:
CIT_DF = CIT_DF_ORIGINAL[CIT_DF_ORIGINAL['Citation Paper OpenAlex ID'].notnull()].iloc[:, 4:]

In [89]:
CIT_DF.shape

(153397, 18)

In [90]:
CIT_DF = CIT_DF.drop_duplicates()

In [91]:
CIT_DF.shape

(60447, 18)

In [92]:
cit_paper_ids = list(set(CIT_DF['Citation Paper OpenAlex ID']))

In [93]:
len(cit_paper_ids)

60274

In [95]:
diff_ids = list(set(CIT_DF['Citation Paper OpenAlex ID']) - set(cit_ids))
diff_ids

['W3211836602', 'W4210838542']

In [96]:
CIT_DF_ORIGINAL[CIT_DF_ORIGINAL[CIT_ID_NAME] == diff_ids[0]]

Cited Ppaer Year              Cited Paper DOI  \
49460              2005  10.1109/INFVIS.2005.1532152   

                 Cited Paper Title Cited Paper OpenAlex ID  OpenAlex Year  \
49460  Interactive Sankey diagrams             W1932144848         2022.0   

      OpenAlex Publication Date Citation Paper OpenAlex ID  \
49460                2022-01-01                W3211836602   

                           Citation Paper OpenAlex Title  \
49460  Investigating the Trinity Between Sustainabili...   

                        Citation Paper OpenAlex DOI  \
49460  https://doi.org/10.1007/978-3-030-79450-7_33   

                             Citation Paper OpenAlex URL  ...  \
49460  https://link.springer.com/chapter/10.1007/978-...  ...   

      OpenAlex Publisher Publication Type OpenAlex First Page  \
49460     Springer, Cham              NaN                 441   

      OpenAlex Last Page Number of References Number of Authors  \
49460                457                 20.0               3.0   

       Number of Concepts  Number of Citations  \
49460                 0.0                  0.0   

                                        Citation API URL  \
49460  https://api.openalex.org/works?filter=cites:W3...   

       Number of Citation API URLs  
49460                          1.0  

[1 rows x 22 columns]

In [97]:
CIT_DF_ORIGINAL[CIT_DF_ORIGINAL[CIT_ID_NAME] == diff_ids[1]]

Cited Ppaer Year            Cited Paper DOI  \
82851               2013      10.1109/TVCG.2013.192   
132648              2017  10.1109/TVCG.2017.2745181   

                                        Cited Paper Title  \
82851   SoccerStories: A Kick-off for Visual Soccer An...   
132648  Bring It to the Pitch: Combining Video and Mov...   

       Cited Paper OpenAlex ID  OpenAlex Year OpenAlex Publication Date  \
82851              W2058878000         2022.0                2022-02-08   
132648             W2753693229         2022.0                2022-02-08   

       Citation Paper OpenAlex ID  \
82851                 W4210838542   
132648                W4210838542   

                            Citation Paper OpenAlex Title  \
82851   The collection, analysis and exploitation of f...   
132648  The collection, analysis and exploitation of f...   

               Citation Paper OpenAlex DOI  \
82851   https://doi.org/10.3233/jsa-200554   
132648  https://doi.org/10.3233/jsa-200554   

               Citation Paper OpenAlex URL  ... OpenAlex Publisher  \
82851   https://doi.org/10.3233/jsa-200554  ...          IOS Press   
132648  https://doi.org/10.3233/jsa-200554  ...          IOS Press   

       Publication Type OpenAlex First Page OpenAlex Last Page  \
82851   journal-article                   1                 37   
132648  journal-article                   1                 37   

       Number of References Number of Authors  Number of Concepts  \
82851                  98.0               3.0                 0.0   
132648                 98.0               3.0                 0.0   

        Number of Citations  \
82851                   0.0   
132648                  0.0   

                                         Citation API URL  \
82851   https://api.openalex.org/works?filter=cites:W4...   
132648  https://api.openalex.org/works?filter=cites:W4...   

        Number of Citation API URLs  
82851                           1.0  
132648                          1.0  

[2 rows x 22 columns]

### Conclusion

It seems that `openalex_citation_paper_df.csv` contains two citation papers that were not included in `openalex_citaiton_concept_df.csv`. I don't know why this happend. 

## I want to look at whether number of citations for each paper is the same as the number of citation papers each VIS paper has in the concept data df. 

In [114]:
df = pd.read_csv('../../data/processed/openalex_paper_df.csv')
df.head(1)

Year                    DOI                     Title  OpenAlex Year  \
0  2011  10.1109/TVCG.2011.185  D³ Data-Driven Documents           2011   

  OpenAlex Publication Date  OpenAlex ID            OpenAlex Title  \
0                2011-12-01  W2135415614  D³ Data-Driven Documents   

                            OpenAlex DOI  \
0  https://doi.org/10.1109/tvcg.2011.185   

                            OpenAlex URL               OpenAlex Venue ID  ...  \
0  https://doi.org/10.1109/tvcg.2011.185  https://openalex.org/V84775595  ...   

  Publication Type OpenAlex First Page OpenAlex Last Page  Number of Pages  \
0  journal-article              2301.0             2309.0              9.0   

   Number of References  Number of Authors  Number of Concepts  \
0                    27                  3                  15   

   Number of Citations                                   Citation API URL  \
0                 2239  https://api.openalex.org/works?filter=cites:W2...   

   Number of Citation API URLs  
0                            1  

[1 rows x 22 columns]

In [117]:
visdoi_citation_count_dic = dict(zip(df.DOI, df['Number of Citations']))

In [118]:
cit_concept_df = pd.read_csv('../../data/processed/large/openalex_citation_concept_df.csv')

In [119]:
cit_concept_df.head(2)

Cited Ppaer Year        Cited Paper DOI         Cited Paper Title  \
0              2011  10.1109/TVCG.2011.185  D³ Data-Driven Documents   
1              2011  10.1109/TVCG.2011.185  D³ Data-Driven Documents   

  Cited Paper OpenAlex ID  Citation Paper Year Citation Paper OpenAlex ID  \
0             W2135415614               2022.0                W4281736408   
1             W2135415614               2022.0                W4281736408   

                       Citation Ppaer OpenAlex Title  \
0  DramatVis Personae: Visual Text Analytics for ...   
1  DramatVis Personae: Visual Text Analytics for ...   

               Citation Paper OpenAlex DOI  Number of Concepts  \
0  https://doi.org/10.1145/3532106.3533526                 8.0   
1  https://doi.org/10.1145/3532106.3533526                 8.0   

   Index of Concept           Concept                      Concept ID  \
0               1.0         Analytics  https://openalex.org/C79158427   
1               2.0  Computer science  https://openalex.org/C41008148   

                                Wikidata  Level     Score  
0  https://www.wikidata.org/wiki/Q485396    2.0  0.704162  
1   https://www.wikidata.org/wiki/Q21198    0.0  0.703784

In [135]:
visdoi_citation_count_dic_to_check = {}
# to keep only cited paper doi and citation paper id. This is because if I simply groupby cited paper doi,
# then there will be multiple entries for each citation paper because this is a concept df!
visdoi_citid_df = cit_concept_df[['Cited Paper DOI', 'Citation Paper OpenAlex ID']]
visdoi_citid_df = visdoi_citid_df.drop_duplicates()
for visdoi, group in visdoi_citid_df.groupby('Cited Paper DOI'):
    visdoi_citation_count_dic_to_check[visdoi] = group.shape[0]

In [138]:
x = visdoi_citation_count_dic 
y = visdoi_citation_count_dic_to_check
shared_items = {k: x[k] for k in x if k in y and x[k] == y[k]}
diff_items = {k: x[k] for k in x if k in y and x[k] != y[k]}
print(len(shared_items))
# What???

2717


In [155]:
print(len(diff_items))

523


In [142]:
diff_visdois = list(diff_items.keys())

In [152]:
diff_df = pd.DataFrame()
diff_df.visdoi = diff_visdois
diff_df['paper_df_cit_count'] = [visdoi_citation_count_dic[x] for x in diff_visdois]
diff_df['concept_df_cit_count'] = [visdoi_citation_count_dic_to_check[x] for x in diff_visdois]
diff_df['difference'] = diff_df['paper_df_cit_count'] - diff_df['concept_df_cit_count']
diff_df['difference_relative'] = abs(diff_df['paper_df_cit_count'] - diff_df['concept_df_cit_count'])

/var/folders/z2/5kr96fyn63z_tj_bwr33t5dw0000gn/T/ipykernel_33131/350390832.py:2: UserWarning: Pandas doesn't allow columns to be created via a new attribute name - see https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute-access
  diff_df.visdoi = diff_visdois


In [154]:
diff_df.difference_relative.sum()

631

In [153]:
diff_df.difference.sum()

157

### Further conclusion

I concluded that this difference is negligible. I guess this is because the API might take some time to update everything. It shows that the total absolute difference in terms of citation counts is 631, occurring in 523 papers. If I sum all the relative differences up, it will be 157, meaning that the total citation counts in paper_df is bigger than that in concept_df. **This is good news. As I used paper_df in citation analysis.**

Also, the above analysis shows that only 2 citation papers in paper_df were not found in concept_df. This is not a big deal at all, as I am analyzing over 60K citation papers' concepts.  

# Checking venue info availability

In [99]:
df[df['OpenAlex Venue Name'].isnull()].shape

(9, 22)

In [100]:
REF_DF.shape

(39817, 19)

In [101]:
REF_DF[REF_DF['OpenAlex Venue Name'].isnull()].shape

(3299, 19)

In [103]:
REF_DF[REF_DF['OpenAlex Venue Name'].isnull()].shape[0]/REF_DF.shape[0]

0.08285405731220333

In [107]:
CIT_DF.shape

(60447, 18)

In [110]:
len(list(set(cit_ids)))

60272

In [104]:
CIT_DF[CIT_DF['OpenAlex Venue Name'].isnull()].shape

(5915, 18)

In [109]:
CIT_DF[CIT_DF['OpenAlex Venue Name'].isnull()].shape[0]/len(list(set(cit_ids)))

0.09813843907618795